### PR TITLE
feat(behavior_path_planner): update filter objects by velocity

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
### <samp>🤖 Generated by Copilot at eeba145</samp>

Modified the `filterObjects` and `filterObjectsByVelocity` functions in `objects_filtering.cpp` and `objects_filtering.hpp` to improve the path safety checker's handling of static or slow-moving objects. Added a new parameter `filter_dynamic_objects` to control the filtering mode based on the object velocity.

filter objects by velocity and if you give arg of `filter_dynamic_objects` as false, object's velocity smaller than `lim_v` will be filtered.  

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by kosuke55 — APPROVED** [review]

File: planning/behavior_path_planner/include/behavior_path_planner/utils/path_safety_checker/objects_filtering.hpp [hunk 1]
```diff
@@ -61,7 +61,8 @@ PredictedObjects filterObjects(
  * @param lim_v Velocity limit for filtering.
  * @return PredictedObjects The filtered objects.
  */
-PredictedObjects filterObjectsByVelocity(const PredictedObjects & objects, double lim_v);
+PredictedObjects filterObjectsByVelocity(
+  const PredictedObjects & objects, const double lim_v, const bool filter_dynamic_objects = true);
```
Comment (kosuke55) [inline planning/behavior_path_planner/include/behavior_path_planner/utils/path_safety_checker/objects_filtering.hpp:65]: `dynamic_objects` means objects whose velocity is larger than lim_v

Comment (kosuke55) [reply planning/behavior_path_planner/include/behavior_path_planner/utils/path_safety_checker/objects_filtering.hpp:65]: a little unclear

Comment (kyoichi-sugahara) [reply planning/behavior_path_planner/include/behavior_path_planner/utils/path_safety_checker/objects_filtering.hpp:65]: @kosuke55 
I agree.
> a little unclear

How about this?
```
/**
 * @brief Filters objects based on their velocity.
 *
 * Depending on the remove_above_threshold parameter, this function either removes objects with
 * velocities above the given threshold or only keeps those objects. It uses the helper function
 * filterObjectsByVelocity() to do the actual filtering.
 *
 * @param objects The objects to be filtered.
 * @param velocity_threshold The velocity threshold for the filtering.
 * @param remove_above_threshold If true, objects with velocities above the threshold are removed.
 *                               If false, only objects with velocities above the threshold are kept.
 * @return A new collection of objects that have been filtered according to the rules.
 */
PredictedObjects filterObjectsByVelocity(
  const PredictedObjects & objects, const double velocity_threshold,
  const bool remove_above_threshold)
  ```

Comment (kosuke55) [reply planning/behavior_path_planner/include/behavior_path_planner/utils/path_safety_checker/objects_filtering.hpp:65]: nice! thanks!!

**Pull request merged**